File: app/src/main/java/es/etologic/mahjongscoring2/data/local_data_source/local/daos/PlayersDao.java
package es.etologic.mahjongscoring2.data.local_data_source.local.daos;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import java.util.List;

import es.etologic.mahjongscoring2.domain.entities.Player;

@Dao
public interface PlayersDao {

    @Insert(onConflict = OnConflictStrategy.FAIL)
    long insert(Player player);

    @Query("SELECT * FROM Players WHERE playerId = :playerId")
    Player getOne(long playerId);

    @Query("SELECT * FROM Players WHERE playerName = :playerName")
    LiveData<Player> getOne(String playerName);

    @Query("SELECT * FROM Players")
    LiveData<List<Player>> getAll();

    @Update
    int updateOne(Player player);

    @Query("DELETE FROM Players WHERE playerId = :playerId")
    int deleteOne(long playerId);

    @Query("DELETE FROM Players")
    int deleteAll();
}

File: app/src/main/java/es/etologic/mahjongscoring2/data/local_data_source/local/daos/PlayersDao.kt
package es.etologic.mahjongscoring2.data.local_data_source.local.daos

import androidx.room.*
import es.etologic.mahjongscoring2.domain.model.Player
import io.reactivex.Single

@Dao
interface PlayersDao {
    
    @Query("SELECT * FROM Players")
    fun getAll(): Single<List<Player>>
    
    @Insert(onConflict = OnConflictStrategy.ABORT)
    fun insert(player: Player): Single<Long>
    
    @Query("SELECT * FROM Players WHERE playerId = :playerId")
    fun getOne(playerId: Long): Single<Player>
    
    @Query("SELECT * FROM Players WHERE playerName = :playerName")
    fun getOne(playerName: String): Single<Player>
    
    @Update
    fun updateOne(player: Player): Single<Int>
    
    @Query("DELETE FROM Players WHERE playerId = :playerId")
    fun deleteOne(playerId: Long): Single<Int>
    
    @Query("DELETE FROM Players")
    fun deleteAll(): Single<Int>
}
